app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt

```diff
@@ -29,61 +31,76 @@ class AdvancedSearchActivity : AppCompatActivity(), DatePickerDialog.OnDateSetLi
         var myMinute: Int = 0
         var timeChanged = false
 
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_pop_up_window)
 
-        val exitPopUpButton = findViewById<Button>(R.id.submit_advanced_search)
-        exitPopUpButton.setOnClickListener {
-            val intent = Intent(this, MainActivity::class.java)
-            intent.putExtra("numAdults", getNumAdults() )
-            intent.putExtra("numChildren", getNumChildren())
-            if (timeChanged){
-                intent.putExtra("time", "$myYear-${makeNumberTwoDigits(myMonth)}-${makeNumberTwoDigits(myDay)}T${makeNumberTwoDigits(myHour)}:${makeNumberTwoDigits(myMinute)}:00.000")
-            } else {
-                intent.putExtra("time", "Unset")
-            }
-            setResult(2, intent)
-            finish()
-        }
+        presenter = AdvancedSearchPresenter()
+        presenter.onViewTaken(this)
 
-        textView = findViewById(R.id.in_date)
-        button = findViewById(R.id.btn_date)
-        button.setOnClickListener {
-            val calendar: Calendar = Calendar.getInstance()
-            day = calendar.get(Calendar.DAY_OF_MONTH)
-            month = calendar.get(Calendar.MONTH)
-            year = calendar.get(Calendar.YEAR)
-            val datePickerDialog =
-                DatePickerDialog(this@AdvancedSearchActivity, this@AdvancedSearchActivity, year, month,day)
-            datePickerDialog.show()
-        }
+        setInitialValues()
+        setListeners()
+
+    }
 
+    private fun setInitialValues() {
```

Comment:
In a way I'd expect this to come from the presenter, in your case it's not really important because it's got very simple initial values, but since the initial values would probably be the same on both platforms I would expect it to come from the shared code?

Reply:
I don't think I'm currently setting mine programatically... I think the goal is in a later branch to add in a recall of previous choices, so if the user were to pick values, accept, and then pick again, then they would start at the values the user just picked.

Reply:
Fair enough, don't worry about it 🙂